src/mlpack/methods/edge_boxes/feature_extraction_impl.hpp

```diff
@@ -0,0 +1,903 @@
+/**
+ * @file feature_extraction_impl.hpp
+ * @author Nilay Jain
+ *
+ * Implementation of feature extraction methods.
+ */
+#ifndef MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_IMPL_HPP
+#define MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_IMPL_HPP
+
+
+#include "feature_extraction.hpp"
+#include <map>
+
+namespace mlpack {
+namespace structured_tree {
+
+template<typename MatType, typename CubeType>
+StructuredForests<MatType, CubeType>::
+StructuredForests(const std::map<std::string, int>& inMap)
+{
+  this->options = inMap;
+}
+
+template<typename MatType, typename CubeType>
+MatType StructuredForests<MatType, CubeType>::
+LoadData(MatType& images, MatType& boundaries, MatType& segmentations)
+{
+  int num_images = this->options["num_images"];
+  int row_size = this->options["row_size"];
+  int col_size = this->options["col_size"];
+  MatType input_data(num_images * row_size * 5, col_size);
+  // we store the input data as follows: 
+  // images (3), boundaries (1), segmentations (1).
+  int loop_iter = num_images * 5;
+  size_t row_idx = 0;
+  int col_i = 0, col_s = 0, col_b = 0;
```

Comment:
use size_t maybe is a better choice, since this type guarantee it is large enough for the index of array